### ISSUE TITLE
Latest devkitPPC GCC Regression

### ISSUE THREAD

**Issue by PsiLupan** [devkitPro/libogc#96]
Have a large project that prior to updating to the latest devkitPro release today from what was presumably GCC 8.2.0, I began experiencing a graphical regression related to the colors when compiling with GCC after updating. 

My project has branch for GCC and another branch for Clang that I regularly switch between, and the issue is not present for Clang. As a result, I don't suspect it's an issue with libogc itself - but it's hard to narrow down. 

I also tested on a Gamecube, and it outright crashes to the stack trace rather than displaying as Dolphin does. Unfortunately, I can't really show that, since my capture card is in a box somewhere and the TV itself is unreadable. This crash not occur for Clang, which renders on a Gamecube correctly.

Not sure I can really come up with a test case here, but figured I'd make note of it in case someone else runs into this.

GCC:
![GCCBug](https://user-images.githubusercontent.com/7443598/85464488-4f9d6800-b56d-11ea-8e4a-bb94b8a9d844.png)

Clang:
![ClangNoBug](https://user-images.githubusercontent.com/7443598/85464468-48765a00-b56d-11ea-8141-1932a10214a9.png)

**Comment by endrift** [comment]
Gonna be hard to diagnose this without code.

**Comment by PsiLupan** [comment]
> Gonna be hard to diagnose this without code.

Yeah, unfortunately, it's too large of a project for me to come up with a small test case here. If I can locate my capture card, I'll at least try to get the stack track that I saw and narrow down at least the difference between Dolphin and console's handling of it that changed with this GCC update.

**Comment by endrift** [comment]
https://github.com/PsiLupan/FRAY is this the project in question? You didn't mention what it was that was breaking.

**Comment by PsiLupan** [comment]
> https://github.com/PsiLupan/FRAY is this the project in question? You didn't mention what it was that was breaking.

It is - Sorry, should've mentioned it.

**Comment by WinterMute** [comment]
-Wno-implicit-function-declaration is a really terrible idea, you really should give the compiler the function signatures it craves and not just turn that warning off.

**Comment by PsiLupan** [comment]
> -Wno-implicit-function-declaration is a really terrible idea, you really should give the compiler the function signatures it craves and not just turn that warning off.

It's a hold over from very, very early on where I was mostly writing code for the sake of getting it written, rather than working piece-by-piece. I had forgotten that was still on, but I'll keep that in mind - Thanks

**Comment by WinterMute** [comment]
Turning that warning back on and adding -Wall I get a lot of warnings that really should be fixed. Implicit function definitions and may be used uninitialized are classic nasal demon triggers.

[build.log](https://github.com/devkitPro/libogc/files/4824141/build.log)

I would be massively unsurprised if fixing all those warnings made the issue go away. Can't guarantee it will but I've seen far too many problems "magically fixed" by simply fixing warnings.
